### PR TITLE
move json-rpc-engine from merged-packages to packages

### LINKED ISSUE #1919: Moving json-rpc-engine from merged-packages into the packages
1. The big leap
 Move migration target from merged-packages/ to packages/.
 Run yarn install in the root directory.
 Check that all tests are passing in migration target by running yarn workspace @metamask/<package-name> test.
2. Update downstream repos
 Add tsconfig reference paths for migration target in downstream packages and root.
 Bump migration target version in downstream packages and root.
Notes on why this version bump needs to happen as part of this PR (import module shadowing): eth-json-rpc-provider migration - Integration into packages/ #1738 (comment)
3. Linter fixes
 Apply yarn constraints fixes to migration target package.json file: yarn constraints --fix (run twice).
 Identify validator fixes for CHANGELOG using yarn workspace @metamask/<package-name> changelog:validate and apply the diffs.
4. Resolve or TODO downstream errors
If introducing the migration target breaks any downstream repos:
 Resolve simple errors
 Mark and ignore complex/blocked errors using @ts-expect-error TODO: annotations.
 Create a separate issue for resolving the marked errors as soon as the migration is completed.
Fix provider, middleware type errors in NetworkController and downstream packages #1823
5. Finalize merge
 Check that all tests are passing in all subpackages of core and CI.
 Merge packages/<package-name> directory into core main branch.

### PR DESCRIPTION
**Pull request by kanthesha**
## Explanation

In the process of migrating json-rpc-engine into core monorepo. In this PR we are moving json-rpc-engine from `merged-packages` to `packages` folder.

In this process, we followed below steps.
Moving to `packages`
1. Move migration target from merged-packages/ to packages/.
2.  Run yarn install in the root directory.
3.  Check that all tests are passing in migration target by running `yarn workspace @metamask/json-rpc-engine test`.

Update downstream repos
1. Bump the json-rpc-engine to 7.2.0 in root and packages
2. Added references in tsconfig.json and tsconfig.build.json

Lints and Constraints
1. Fixed `yarn constraints` errors
2. Applied `yarn workspace @metamask/json-rpc-engine changelog:validate`

## References

- Closes #1919 

### `@metamask/eth-json-rpc-provider`

- **CHANGED:** Bump dependency on `@metamask/json-rpc-engine` to ^7.2.0

### `@metamask/json-rpc-engine`

- **CHANGED:** Bump dependency on `@metamask/utils` to ^8.2.0

### `@metamask/network-controller`

- **CHANGED:** Bump dependency on `@metamask/json-rpc-engine` to ^7.2.0

### `@metamask/permission-controller`

- **CHANGED:** Bump dependency on `@metamask/json-rpc-engine` to ^7.2.0

### `@metamask/queued-request-controller`

- **CHANGED:** Bump dependency on `@metamask/json-rpc-engine` to ^7.2.0

### `@metamask/selected-network-controller`

- **CHANGED:** Bump dependency on `@metamask/json-rpc-engine` to ^7.2.0

## Checklist

- [ ] I've updated the test suite for new or updated code as appropriate
- [ ] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [ ] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

File: packages/json-rpc-engine/CHANGELOG.md [hunk 1]
```diff
@@ -83,13 +83,13 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+[Unreleased]: https://github.com/MetaMask/core/compare/@metamask/json-rpc-engine@7.2.0...HEAD
+[7.2.0]: https://github.com/MetaMask/core/compare/@metamask/json-rpc-engine@7.1.1...@metamask/json-rpc-engine@7.2.0
+[7.1.1]: https://github.com/MetaMask/core/compare/@metamask/json-rpc-engine@7.1.0...@metamask/json-rpc-engine@7.1.1
+[7.1.0]: https://github.com/MetaMask/core/compare/@metamask/json-rpc-engine@7.0.0...@metamask/json-rpc-engine@7.1.0
+[7.0.0]: https://github.com/MetaMask/core/compare/@metamask/json-rpc-engine@6.1.0...@metamask/json-rpc-engine@7.0.0
+[6.1.0]: https://github.com/MetaMask/core/compare/@metamask/json-rpc-engine@6.0.0...@metamask/json-rpc-engine@6.1.0
+[6.0.0]: https://github.com/MetaMask/core/compare/@metamask/json-rpc-engine@5.4.0...@metamask/json-rpc-engine@6.0.0
+[5.4.0]: https://github.com/MetaMask/core/compare/@metamask/json-rpc-engine@5.3.0...@metamask/json-rpc-engine@5.4.0
+[5.3.0]: https://github.com/MetaMask/core/compare/@metamask/json-rpc-engine@5.2.0...@metamask/json-rpc-engine@5.3.0
+[5.2.0]: https://github.com/MetaMask/core/releases/tag/@metamask/json-rpc-engine@5.2.0
```
Comment (MajorLift) [inline packages/json-rpc-engine/CHANGELOG.md:95]: The `--version-before-package-rename` for `json-rpc-engine` appears to be `6.1.0`. Does the `changelog:validate` run show a different result?

```suggestion
[Unreleased]: https://github.com/MetaMask/core/compare/@metamask/json-rpc-engine@7.2.0...HEAD
[7.2.0]: https://github.com/MetaMask/core/compare/@metamask/json-rpc-engine@7.1.1...@metamask/json-rpc-engine@7.2.0
[7.1.1]: https://github.com/MetaMask/core/compare/@metamask/json-rpc-engine@7.1.0...@metamask/json-rpc-engine@7.1.1
[7.1.0]: https://github.com/MetaMask/core/compare/@metamask/json-rpc-engine@7.0.0...@metamask/json-rpc-engine@7.1.0
[7.0.0]: https://github.com/MetaMask/core/compare/@metamask/json-rpc-engine@6.1.0...@metamask/json-rpc-engine@7.0.0
[6.1.0]: https://github.com/MetaMask/core/compare/json-rpc-engine@6.0.0...json-rpc-engine@6.1.0
[6.0.0]: https://github.com/MetaMask/core/compare/json-rpc-engine@5.4.0...json-rpc-engine@6.0.0
[5.4.0]: https://github.com/MetaMask/core/compare/json-rpc-engine@5.3.0...json-rpc-engine@5.4.0
[5.3.0]: https://github.com/MetaMask/core/compare/json-rpc-engine@5.2.0...json-rpc-engine@5.3.0
[5.2.0]: https://github.com/MetaMask/core/releases/tag/json-rpc-engine@5.2.0
```

**Review by MajorLift — CHANGES_REQUESTED** [review]
I've finished testing the tag porting script, and I think it will be safer to run it before the package is moved out of `merged-packages/`. 

So https://github.com/MetaMask/core/pull/1802 will need to be merged before this PR can be merged. Sorry for the confusion!

File: packages/json-rpc-engine/CHANGELOG.md (hunk 1 above)
Comment (MajorLift) [reply packages/json-rpc-engine/CHANGELOG.md:95]: The rename to `@metamask/` namespace appears to have taken place at v7.0.0 according to the [`CHANGELOG`](https://github.com/MetaMask/json-rpc-engine/compare/v6.1.0...v7.0.0#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4edR20) and [`package.json`](https://github.com/MetaMask/json-rpc-engine/compare/v6.1.0...v7.0.0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R2-R3) from the v6.1.0...v7.0.0 diff.

This is the PR where the package rename took place: https://github.com/MetaMask/json-rpc-engine/pull/139/files. 

Is there maybe a second rename that happened or something else I'm missing?

**Review by MajorLift — DISMISSED** [review]
Tags have been successfully ported to core! 
- Example tag diff link from CHANGELOG: https://github.com/MetaMask/core/compare/json-rpc-engine@6.1.0...@metamask/json-rpc-engine@7.0.0
- Verify full list with this command:
```bash
git ls-remote --tags origin | grep 'json-rpc-engine'
```

**Comment by MajorLift** [comment]
Note on tags: Some earlier release commits that were untagged in the original repo have been skipped by the tag porting script, and will be tagged manually.

- TODO: GitHub may show these manually-created tags out of order due to sorting by tag creation date instead of release commit timestamp.

**Review by FrederikBolding — APPROVED** [review]
Reviewed PermissionController diff

File: packages/permission-controller/package.json [hunk 2]
```diff
@@ -33,9 +33,9 @@
     "@metamask/approval-controller": "^4.1.0",
     "@metamask/base-controller": "^3.2.3",
     "@metamask/controller-utils": "^5.0.2",
-    "@metamask/json-rpc-engine": "^7.1.1",
+    "@metamask/json-rpc-engine": "^7.2.0",
     "@metamask/rpc-errors": "^6.1.0",
-    "@metamask/utils": "^8.2.0",
```
Comment (FrederikBolding) [inline packages/permission-controller/package.json:38]: Why is this package being downgraded?

Comment (MajorLift) [reply packages/permission-controller/package.json:38]: Thanks for catching this! I just pushed a fix. 

I'll add an entry in the migration checklist to double-check for dependency version bumps or other changes in main that may have happened while the PR is open. 

For future migrations, we plan to coordinate to minimize the time these PRs are kept open so there's less chance of something like this slipping through.

**Review by mcmire — APPROVED** [review]
I updated the PR description to mention the dependency changes to other packages being made here. This looks good.

**Review by FrederikBolding — APPROVED** [review]

**Pull request merged**